dan_layer/engine_types/src/resource_container.rs

```diff
@@ -187,6 +188,20 @@ impl ResourceContainer {
             )),
         }
     }
+
+    pub fn withdraw_confidential(&mut self, _proofs: &[ConfidentialProof]) -> Result<ResourceContainer, ResourceError> {
```

Comment:
I think you should only be allowed to withdraw the full commitment and use Split to get change and deposit back